src/algebra/pointwise.lean

```diff
@@ -734,27 +734,49 @@ protected def has_mul [has_mul α] : has_mul (finset α) :=
 localized "attribute [instance] finset.has_mul finset.has_add" in pointwise
 
 @[to_additive]
-lemma mul_def [has_mul α] {s t : finset α} :
+lemma mul_def [has_mul α] :
   s * t = (s.product t).image (λ p : α × α, p.1 * p.2) := rfl
 
 @[to_additive]
-lemma mem_mul [has_mul α] {s t : finset α} {x : α} :
+lemma mem_mul [has_mul α] {x : α} :
   x ∈ s * t ↔ ∃ y z, y ∈ s ∧ z ∈ t ∧ y * z = x :=
 by { simp only [finset.mul_def, and.assoc, mem_image, exists_prop, prod.exists, mem_product] }
 
 @[simp, norm_cast, to_additive]
-lemma coe_mul [has_mul α] {s t : finset α} : (↑(s * t) : set α) = ↑s * ↑t :=
+lemma coe_mul [has_mul α] : (↑(s * t) : set α) = ↑s * ↑t :=
 by { ext, simp only [mem_mul, set.mem_mul, mem_coe] }
 
 @[to_additive]
-lemma mul_mem_mul [has_mul α] {s t : finset α} {x y : α} (hx : x ∈ s) (hy : y ∈ t) :
+lemma mul_mem_mul [has_mul α] {x y : α} (hx : x ∈ s) (hy : y ∈ t) :
   x * y ∈ s * t :=
 by { simp only [finset.mem_mul], exact ⟨x, y, hx, hy, rfl⟩ }
 
 @[to_additive]
-lemma mul_card_le [has_mul α] {s t : finset α} : (s * t).card ≤ s.card * t.card :=
+lemma mul_card_le [has_mul α] : (s * t).card ≤ s.card * t.card :=
 by { convert finset.card_image_le, rw [finset.card_product, mul_comm] }
 
+@[simp, to_additive] lemma empty_mul [has_mul α] : ∅ * s = ∅ :=
+eq_empty_of_forall_not_mem (by simp [mem_mul])
+
+@[simp, to_additive] lemma mul_empty [has_mul α] : s * ∅ = ∅ :=
+eq_empty_of_forall_not_mem (by simp [mem_mul])
```

Comment:
`s` should be explicit to these lemmas, and probably to a few lemmas above too, although those were already there so fixing them is out of scope if you want it to be.